media-gfx/digikam/digikam-5.3.0.ebuild

```diff
@@ -0,0 +1,166 @@
+	$(add_frameworks_dep kxmlgui)
+	$(add_frameworks_dep solid)
+	$(add_qt_dep qtdbus)
+	$(add_qt_dep qtgui '-gles2')
+	$(add_qt_dep qtprintsupport)
+	$(add_qt_dep qtsql 'mysql?')
+	$(add_qt_dep qtwebkit)
+	$(add_qt_dep qtwidgets)
+	$(add_qt_dep qtxml)
+	dev-libs/expat
+	media-gfx/exiv2:=
+	media-libs/lcms:2
+	media-libs/liblqr
+	>=media-libs/libpgf-6.12.27
+	media-libs/libpng:0=
+	media-libs/opencv:=[-qt4]
+	|| ( <media-libs/opencv-3.0.0 >=media-libs/opencv-3.1.0 )
+	media-libs/tiff:0
+	virtual/jpeg:0
+	addressbook? (
+		$(add_kdeapps_dep akonadi-contacts)
+		$(add_kdeapps_dep kcontacts)
+	)
+	calendar? ( $(add_kdeapps_dep kcalcore) )
+	gphoto2? ( media-libs/libgphoto2:= )
+	jpeg2k? ( media-libs/jasper )
+	kipi? ( $(add_kdeapps_dep libkipi '' '16.03.80') )
+	lensfun? ( media-libs/lensfun )
+	marble? (
+		$(add_frameworks_dep kbookmarks)
+		$(add_kdeapps_dep marble)
+		$(add_qt_dep qtconcurrent)
+		$(add_qt_dep qtnetwork)
+	)
+	mysql? ( virtual/mysql )
+	opengl? (
+		$(add_qt_dep qtopengl)
+		virtual/opengl
+	)
+	panorama? ( $(add_frameworks_dep threadweaver) )
+	scanner? ( $(add_kdeapps_dep libksane) )
+	semantic-desktop? ( $(add_frameworks_dep kfilemetadata) )
+	video? ( $(add_qt_dep qtmultimedia 'widgets') )
+	X? (
+		$(add_qt_dep qtx11extras)
+		x11-libs/libX11
+	)
+"
+DEPEND="${COMMON_DEPEND}
+	$(add_qt_dep qtconcurrent)
+	dev-cpp/eigen:3
+	dev-libs/boost[threads]
+	sys-devel/gettext
+	panorama? (
+		sys-devel/bison
+		sys-devel/flex
+	)
+"
+RDEPEND="${COMMON_DEPEND}
+	media-plugins/kipi-plugins:5
+	panorama? ( media-gfx/hugin )
+	!media-gfx/digikam:4
+"
+
+RESTRICT=test
+# bug 366505
+
+# FIXME: Unbundle libraw (libs/rawengine/libraw)
+pkg_pretend() {
+	if use openmp ; then
+		tc-has-openmp || die "Please switch to an openmp compatible compiler"
+	fi
+	kde5_pkg_pretend
+}
+
+src_prepare() {
+	if [[ ${KDE_BUILD_TYPE} != live ]]; then
+		# prepare the translations
+		mv "${WORKDIR}/${MY_P}/po" po || die
+		find po -name "*.po" -and -not -name "digikam.po" -delete || die
+		echo "set_property(GLOBAL PROPERTY ALLOW_DUPLICATE_CUSTOM_TARGETS 1)" >> CMakeLists.txt || die
+		echo "find_package(Gettext REQUIRED)" >> CMakeLists.txt || die
+		echo "add_subdirectory( po )" >> CMakeLists.txt || die
+
+		if use handbook; then
+			# subdirs need to be preserved b/c relative paths...
+			# doc-translated is, in fact, broken, and ignored
+			mv "${WORKDIR}/${MY_P}/doc/${PN}" doc-default || die
+			echo "find_package(KF5DocTools REQUIRED)" >> CMakeLists.txt || die
+			echo "add_subdirectory( doc-default )" >> CMakeLists.txt || die
+		fi
+	fi
+
+	if ! use marble; then
+		punt_bogus_dep Qt5 Network
+	fi
+
+	kde5_src_prepare
+}
+
+src_configure() {
+	# LQR = only allows to choose between bundled/external
+	local mycmakeargs=(
+		-DENABLE_APPSTYLES=ON
+		-DENABLE_AKONADICONTACTSUPPORT=$(usex addressbook)
+		-DENABLE_MYSQLSUPPORT=$(usex mysql)
+		-DENABLE_MEDIAPLAYER=$(usex video)
+		-DENABLE_OPENCV3=$(has_version ">=media-libs/opencv-3" && echo yes || echo no)
```

Comment:
This seems brittle. compile-time properties depend on an `RDEPEND` that can in theory be replaced later on. Is it such a problem just hard depending on OpenCV 3 and later?